src/lib.rs

```diff
@@ -1,129 +1,155 @@
+            zed::Os::Linux => ("unknown-linux-gnu", "tar.gz"),
+            zed::Os::Windows => ("pc-windows-msvc", "zip"),
+        };
 
-        let gh_asset = release
+        let asset_name = format!("harper-ls-{arch_name}-{os_str}.{ext}");
+        let asset = release
             .assets
             .iter()
-            .find(|asset| asset.name == asset_name)
-            .ok_or_else(|| format!("no asset found matching {:?}", asset_name))?;
+            .find(|a| a.name == asset_name)
+            .ok_or_else(|| format!("No compatible Harper binary found for {arch_name}-{os_str}"))?;
 
-        let version_dir = format!("harper-ls-{}", release.version);
-        let binary_path = format!("{version_dir}/harper-ls");
+        let version_dir = PathBuf::from(format!("harper-ls-{}", release.version));
+        let mut binary_path = version_dir.join("harper-ls");
 
-        if !fs::metadata(&binary_path).map_or(false, |stat| stat.is_file()) {
+        if platform == zed::Os::Windows {
+            binary_path.set_extension("exe");
+        }
+
+        if !binary_path.exists() {
             zed::set_language_server_installation_status(
                 language_server_id,
                 &zed::LanguageServerInstallationStatus::Downloading,
             );
 
             zed::download_file(
-                &gh_asset.download_url,
-                &version_dir,
-                zed::DownloadedFileType::GzipTar,
+                &asset.download_url,
+                version_dir
+                    .to_str()
+                    .ok_or("Invalid version directory path")?,
+                if cfg!(windows) {
+                    zed::DownloadedFileType::Zip
+                } else {
+                    zed::DownloadedFileType::GzipTar
+                },
             )
-            .map_err(|e| format!("failed to download file: {e}"))?;
+            .map_err(|e| format!("Failed to download Harper binary: {}", e))?;
 
-            zed::make_file_executable(&binary_path)?;
+            zed::make_file_executable(
+                binary_path
+                    .to_str()
+                    .ok_or("Failed to convert binary path to string")?,
+            )
+            .map_err(|e| format!("Failed to make binary executable: {}", e))?;
 
-            let entries =
-                fs::read_dir(".").map_err(|e| format!("failed to list working directory {e}"))?;
-            for entry in entries {
-                let entry = entry.map_err(|e| format!("failed to load directory entry {e}"))?;
-                if entry.file_name().to_str() != Some(&version_dir) {
-                    fs::remove_dir_all(entry.path()).ok();
+            if let Ok(entries) = fs::read_dir(".") {
+                for entry in entries.flatten() {
+                    if entry.path() != version_dir {
+                        fs::remove_dir_all(entry.path()).ok();
+                    }
                 }
             }
```

Comment:
This change contains a regression for MacOS. On a fresh install, this new approach will remove the binary location. I haven't looked into why. The version dir will still exist, just without the LSP binary in it.

If this is a fix you made for Windows we'll have to investigate how to fix it for other platforms, if it's purely stylistically I prefer to revert it.

Reply:
Im pretty sure we can revert it. this is just stylistically and will not do anything for the windows fix

Reply:
Alright so i pushed a fix, could you test the extension now?
I do not have a access to a mac, but i don't think this shouldn't fail now.

Reply:
Tested, works thanks!

Reply:
Waiting for merge in extensions repo: https://github.com/zed-industries/extensions/pull/1699